src/screens/SignTransaction/02-ConnectDevice.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import invariant from "invariant";
+import React, { useMemo } from "react";
+import { StyleSheet } from "react-native";
+import { useSelector } from "react-redux";
+import SafeAreaView from "react-native-safe-area-view";
+import { getMainAccount } from "@ledgerhq/live-common/lib/account";
+import type {
+  Transaction,
+  TransactionStatus,
+} from "@ledgerhq/live-common/lib/types";
+import { createAction } from "@ledgerhq/live-common/lib/hw/actions/transaction";
+import connectApp from "@ledgerhq/live-common/lib/hw/connectApp";
+import type { Device } from "@ledgerhq/live-common/lib/hw/actions/types";
+import useBridgeTransaction from "@ledgerhq/live-common/lib/bridge/useBridgeTransaction";
+import { useTheme } from "@react-navigation/native";
+import { accountScreenSelector } from "../../reducers/accounts";
+import DeviceAction from "../../components/DeviceAction";
+import { TrackScreen } from "../../analytics";
+import { useSignedTxHandlerWithoutBroadcast } from "../../logic/screenTransactionHooks";
+
+const action = createAction(connectApp);
+
+type Props = {
+  navigation: any,
+  route: {
+    params: RouteParams,
+    name: string,
+  },
+};
+
+type RouteParams = {
+  device: Device,
+  accountId: string,
+  transaction: Transaction,
+  status: TransactionStatus,
+  appName?: string,
+  onSuccess: (payload: *) => void,
+  onError: (error: Error) => void,
+};
+
+export default function ConnectDevice({ route }: Props) {
+  const { colors } = useTheme();
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+  invariant(account, "account is required");
+
+  const { appName, onSuccess } = route.params;
+
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  const { transaction, status } = useBridgeTransaction(() => ({
+    account: mainAccount,
+    transaction: route.params.transaction,
+  }));
+
+  const tokenCurrency =
+    account.type === "TokenAccount" ? account.token : undefined;
+
+  const handleTx = useSignedTxHandlerWithoutBroadcast({
+    onSuccess,
+  });
+
+  return useMemo(
```

Comment:
useMemo for React Element feels a new/weird pattern to me. but maybe i'm just not used to it!
it still feels a bit dangerous pattern, specifically combining the use of react-hooks/exhaustive-deps makes it easy to mess things up.

I think what you want is instead to freeze the `request` prop into a object. (it's that request prop that needs to be memo-ed)

Reply:
This is a copy/pasta from the `screens/ConnectDevice`, that should maybe be investigated and fixed.
We can try to add a React.memo on our component, but i'm reluctant to update components out of our scope.
This might not even be useful here.

Reply:
Not sure if that's enough, but I changed it using this pattern recommended by @LFBarreto :
```
import React, { memo } from "react";
function Component({}: Props) {...}
export default memo<Props>(Component);
```